api/python/provisioner/commands/__init__.py

```diff
@@ -808,6 +808,10 @@ def run(self, targets):  # noqa: C901 FIXME
                 try:
                     _update_component('provisioner', targets)
 
+                    # re-apply provisioner configuration to ensure
+                    # that updated pillar is taken into account
+                    _apply_provisioner_config(targets)
```

Comment:
@andkononykhin2 
We also need to update `components.provisioner.config` to refresh pillar, which is being missed before encryption call `components.system.config.pillar_encrypt` (essentially missed in the CSM secret update case)

Reply:
> @andkononykhin2 
> 
> We also need to update `components.provisioner.config` to refresh pillar, which is being missed before encryption call `components.system.config.pillar_encrypt` (essentially missed in the CSM secret update case)

My doubts were cleared during our discussion.

Reply:
Thanks